spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -138,7 +140,10 @@ BatchWrite asBatchAppend() {
   }
 
   BatchWrite asDynamicOverwrite() {
-    return new DynamicOverwrite();
+    String isolationLevelAsString = table.properties().getOrDefault(REPLACE_PARTITION_LEVEL,
+        REPLACE_PARTITION_LEVEL_DEFAULT);
+    IsolationLevel level = IsolationLevel.valueOf(isolationLevelAsString.toUpperCase(Locale.ROOT));
+    return new DynamicOverwrite(level, table.currentSnapshot().snapshotId());
```

Comment:
You might consider doing this differently. Maybe accept a write option that sets the starting snapshot ID for now? Defaulting this to the current snapshot as of the time that the dynamic overwrite is created seems unreliable to me.

Reply:
Got it, I implemented the suggestion (taking your example from https://github.com/apache/iceberg/pull/3069#discussion_r711803863), and added a Spark level unit test